src/components/taskDetails/Details.tsx

```diff
@@ -68,34 +87,52 @@ const Details: FC<TaskDetailsProps> = ({ detailType, value, url }) => {
-                style={{ color: color ?? 'black' }}
-            >
-                {isGitHubLink && value ? (
-                    <a
-                        className={styles.gitLink}
-                        href={gitHubIssueLink}
-                        target="_blank"
-                        rel="noopener noreferrer"
-                        aria-label="Open GitHub Issue"
-                        title={value}
-                    >
-                        {isGitHubLink ? `${extractRepoName(value)}` : value}
-                    </a>
-                ) : isTimeDetail ? (
-                    <Tooltip
-                        content={formatDate(value)}
-                        tooltipPosition={{ top: '-3.6rem', right: '-4.5rem' }}
-                    >
-                        {tooltipActive
-                            ? formatDate(value)
-                            : getRelativeTime(value)}
-                    </Tooltip>
-                ) : (
-                    renderedValue
-                )}
-            </span>
+            {isEditing && isUserAuthorized ? (
+                <input
+                    data-testid="endsOnTaskDetails"
+                    type="date"
+                    name="endsOn"
+                    onChange={(e) => setNewEndOnDate(e.target.value)}
+                    onBlur={handleBlurOfEndsOn}
+                    value={
+                        newEndOnDate ||
+                        new Date(value as string).toLocaleDateString('en-CA')
+                    }
+                    className={styles.inputField}
+                />
+            ) : (
+                <span
+                    className={styles.detailValue}
+                    style={{ color: color ?? 'black' }}
+                >
+                    {isGitHubLink && value ? (
+                        <a
+                            className={styles.gitLink}
+                            href={gitHubIssueLink}
+                            target="_blank"
+                            rel="noopener noreferrer"
+                            aria-label="Open GitHub Issue"
+                            title={value}
+                        >
+                            {isGitHubLink ? `${extractRepoName(value)}` : value}
+                        </a>
+                    ) : isTimeDetail ? (
+                        <Tooltip
+                            content={formatDate(value)}
+                            tooltipPosition={{
+                                top: '-3.6rem',
+                                right: '-4.5rem',
+                            }}
+                        >
+                            {tooltipActive
+                                ? formatDate(value)
+                                : getRelativeTime(value)}
+                        </Tooltip>
+                    ) : (
+                        renderedValue
+                    )}
+                </span>
+            )}
```

Comment:
Why too much nesting?

Reply:
I know, actually I saw it being already there, do you want me to refactor in this PR only?

Reply:
Then make your timedetails a component in same file

Reply:
okay

Reply:
updated